openpype/hosts/webpublisher/plugins/publish/collect_published_files.py

```diff
@@ -0,0 +1,266 @@
+"""Loads publishing context from json and continues in publish process.
+
+Requires:
+    anatomy -> context["anatomy"] *(pyblish.api.CollectorOrder - 0.11)
+
+Provides:
+    context, instances -> All data from previous publishing process.
+"""
+
+import os
+import json
+import clique
+
+import pyblish.api
+from avalon import io
+from openpype.lib import prepare_template_data
+
+
+class CollectPublishedFiles(pyblish.api.ContextPlugin):
+    """
+    This collector will try to find json files in provided
+    `OPENPYPE_PUBLISH_DATA`. Those files _MUST_ share same context.
+
+    """
+    # must be really early, context values are only in json file
+    order = pyblish.api.CollectorOrder - 0.490
+    label = "Collect rendered frames"
+    host = ["webpublisher"]
+
+    _context = None
+
+    # from Settings
+    task_type_to_family = {}
+
+    def _load_json(self, path):
+        path = path.strip('\"')
+        assert os.path.isfile(path), (
+            "Path to json file doesn't exist. \"{}\"".format(path)
+        )
+        data = None
+        with open(path, "r") as json_file:
+            try:
+                data = json.load(json_file)
+            except Exception as exc:
+                self.log.error(
+                    "Error loading json: "
+                    "{} - Exception: {}".format(path, exc)
+                )
+        return data
+
+    def _process_batch(self, dir_url):
+        task_subfolders = [
+            os.path.join(dir_url, o)
+            for o in os.listdir(dir_url)
+            if os.path.isdir(os.path.join(dir_url, o))]
+        self.log.info("task_sub:: {}".format(task_subfolders))
+        for task_dir in task_subfolders:
+            task_data = self._load_json(os.path.join(task_dir,
+                                                     "manifest.json"))
+            self.log.info("task_data:: {}".format(task_data))
+            ctx = task_data["context"]
+            task_type = "default_task_type"
+            task_name = None
+
+            if ctx["type"] == "task":
+                items = ctx["path"].split('/')
+                asset = items[-2]
+                os.environ["AVALON_TASK"] = ctx["name"]
```

Comment:
Is there a possibility that there are more than one different tasks in one manifest? Because it would make more sense to set `"task"` key on instance rather than environment which is overriden on each iterration of `task_subfolders`.

Reply:
No, there is always at most single task per batch (manifest).